content/posts/a-guide-on-commit-messages.mdx

```diff
@@ -0,0 +1,169 @@
+
+Taking care of your commit messages can also simplify your daily life. Some tools becomes more interesting with a meaningful commit history like a few git commands `log`, `rebase`, `cherry-pick`, ... When you come back from holidays a simple `git log --pretty` will give you a lot of information about what your coworkers did while you was away. Writing good commit messages force you to split your change making it easier to review with `git log -p`.
+
+Speaking of review did you ever struggle rebasing your code on some change a developer made? It becomes easier to know why a change was made if the commit message carries real meaning. It makes `rebase` result much more predictible. Another example is `git blame` command which is by the way not made to make fun of your colleagues but giving you context on a line of code you come accross. It can be pretty useful when you want to refactor a piece of code. 
+
+Last but not least, a well structured commit history enables generating a changelog between two version of your project to inform your users what changed.
+
+This is a huge amount of reasons to start taking time to write good commit messages.
+
+## How to write a good commit message
+
+Now that you are convinced commit messages are important. Let's talk about how to write good ones. I picked commit messages from a project I work on called [immutadot](https://github.com/zenika-open-source/immutadot). As you can see these commit messages are not consistent. It makes them hard to read.
+
+```
+fix #72 unnecessary object copies
+Fixing unshift documentation examples
+seq.chain path parameter fix #18
+Remove built files
+Add tests
+Add Circle CI base config
+Rename es -> src
+Add lang package with toggle, fix exports
+Core package
+```
+
+In these commit message we can see a few mistakes in term of style, content and metadata. Chris beams describe well these three points in his article [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
+> Style. Markup syntax, wrap margins, grammar, capitalization, punctuation. Spell these things out, remove the guesswork, and make it all as simple as possible. The end result will be a remarkably consistent log that’s not only a pleasure to read but that actually does get read on a regular basis.
+> 
+> Content. What kind of information should the body of the commit message (if any) contain? What should it not contain?
+> 
+> Metadata. How should issue tracking IDs, pull request numbers, etc. be referenced?
+> 
+> Chris Beams - [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+We can see sometime a capitalized message. There are metadata for only two commits. Finally some messages content are so light we can't figure out what the commit does. In the same article Chris goes through 7 rules to write a good commit message.
```

Comment:
```suggestion
We can see there are capitalized messages, metadata for only two commits... Eventually, some messages content are so light we can't figure out what the commit does. In the same article, Chris goes through 7 rules to write a good commit message :
```

Reply:
There is not space before `:`, no? Like for other punctuation characters.